### ISSUE TITLE
TST: Tests broken with semantic versioning

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#9186]
Example log: https://travis-ci.org/astropy/astropy/jobs/578463263

```
______________ ERROR collecting astropy/io/misc/asdf/extension.py ______________
astropy/io/misc/asdf/extension.py:21: in <module>
    from .tags.time.time import *
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
__ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py __
astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py:16: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:21: in <module>
    from .tags.time.time import *
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
__ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py __
astropy/io/misc/asdf/tags/coordinates/tests/test_angle.py:16: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:21: in <module>
    from .tags.time.time import *
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
_ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py __
astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py:14: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:21: in <module>
    from .tags.time.time import *
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
_ ERROR collecting astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py __
astropy/io/misc/asdf/tags/coordinates/tests/test_frames.py:14: in <module>
    from astropy.io.misc.asdf.extension import AstropyExtension
astropy/io/misc/asdf/extension.py:21: in <module>
    from .tags.time.time import *
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
___________ ERROR collecting astropy/io/misc/asdf/tags/time/time.py ____________
astropy/io/misc/asdf/tags/time/time.py:35: in <module>
    class TimeType(AstropyAsdfType):
astropy/io/misc/asdf/tags/time/time.py:38: in TimeType
    supported_versions = ['1.0.0', AsdfSpec('>=1.1.0')]
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/asdf/versioning.py:129: in __init__
    super(AsdfSpec, self).__init__(*args, **kwargs)
/home/travis/miniconda/envs/test/lib/python3.7/site-packages/semantic_version/base.py:499: in __init__
    stacklevel=2,
E   DeprecationWarning: The `SpecItem` class will be removed in 3.0.
!!!!!!!!!!!!!!!!!!! Interrupted: 6 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 6 error in 30.98 seconds ===========================
```

**Comment by nden** [comment]
I'm working on a fix.

**Comment by nden** [comment]
Fixed by asdf 2.4.2 release.
